### PR TITLE
fix(levm): change gas to return in `post_execution_changes`

### PR DESCRIPTION
**Pull request by damiramirez**
**Motivation**

We have a problem when we need to refund the senders gas but it is too high.

**Description**

This PR address an issue where we fail with `InternalState` error due to an overflow when multiplying a very large number of `gas_to_return` by `gas_price`. The overflow occur because the `u64` type cannot handle the new value. 

I change the name of the variable to something that is more related to what it does: `gwei_return_amount`. Since this gas is what we are going to add to the balance, it converts to `gwei` and we use `U256`

```rust
let gwei_return_amount = self
            .env
            .gas_price
            .checked_mul(U256::from(gas_to_return))
            .ok_or(VMError::Internal(InternalError::UndefinedState(1)))?;
self.increase_account_balance(sender_address, gwei_return_amount)?;
```

### CODE REVIEW

File: crates/vm/levm/src/vm.rs [hunk 1]
```diff
@@ -709,14 +709,13 @@ impl VM {
             .and_then(|gas| gas.checked_add(refunded_gas))
             .ok_or(VMError::Internal(InternalError::UndefinedState(0)))?;
 
-        let gas_return_amount = self
+        let gwei_return_amount = self
```
Comment (LeanSerra) [inline crates/vm/levm/src/vm.rs:712]: I think the value is measured in wei

File: crates/vm/levm/src/vm.rs [hunk 2]
```diff
@@ -709,14 +709,13 @@ impl VM {
...
             .env
             .gas_price
-            .low_u64()
-            .checked_mul(gas_to_return)
+            .checked_mul(U256::from(gas_to_return))
             .ok_or(VMError::Internal(InternalError::UndefinedState(1)))?;
 
-        self.increase_account_balance(sender_address, U256::from(gas_return_amount))?;
+        self.increase_account_balance(sender_address, gwei_return_amount)?;
```
Comment (LeanSerra) [inline crates/vm/levm/src/vm.rs:718]: Ditto

**Review by LeanSerra — CHANGES_REQUESTED** [review]
LGTM but the variable name is wrong

File: crates/vm/levm/src/vm.rs (hunk 1 above)
Comment (JereSalo) [reply crates/vm/levm/src/vm.rs:712]: Yeah. With that changed we can merge

Comment (damiramirez) [reply crates/vm/levm/src/vm.rs]: Done in [ee33452](https://github.com/lambdaclass/ethrex/pull/1464/commits/ee33452a232dc510dd4332edb5df43fdf44dfc92)

File: crates/vm/levm/src/vm.rs (hunk 2 above)
Comment (damiramirez) [reply crates/vm/levm/src/vm.rs]: Done in [ee33452](https://github.com/lambdaclass/ethrex/pull/1464/commits/ee33452a232dc510dd4332edb5df43fdf44dfc92)

**Review by JereSalo — COMMENTED** [review]
I was about to merge but I was thinking that this PR can also involve change of the same for the coinbase address, it is a few lines below. We can change the wei to send to the coinbase to `U256`

**Review by JereSalo — APPROVED** [review]
lgtm

**Review by LeanSerra — APPROVED** [review]
LGTM

**Pull request merged**